### DISCUSSION TITLE
How can i securely integrate 3rd party services with Github?

### DISCUSSION THREAD

**Discussion by mrepol742** [community/community#57855] [Actions] [answered]
### Select Topic Area

Question

### Body

How can i securely integrate 3rd party services with Github?

**Comment by imbharat420** [comment, marked as answer]
Their are some best practices to integrate third party services like:

- Use O-Auth token based authentication instead sharing credentials to third pary

- Only grant necessary scope permissions

- Enable 2fa 

- use webhooks with payload verifications

- Monitor and review access permissions regularly and audit permission granted to  third oarty services within your GitHub account and revoke access when no longer needed

-  Regularly rotate access token this reduces the risk of unauthorised access in case token is compromised

**Comment by esparter** [comment]
How does that work with 2FA?
Github documentation says:
"**The only way** someone can sign into your account is if they know both your password and have access to the authentication code on your phone.
How can my CI server login as it does not have access to my phone?
Or is this documentation wrong?
